wire-schema/src/main/java/com/squareup/wire/schema/SchemaLoader.java

```diff
@@ -45,38 +42,26 @@
  * relative to the root of the archive.
  */
 public final class SchemaLoader {
-  private static final Function<File, Path> FILE_TO_PATH = new Function<File, Path>() {
-    @Override public Path apply(File file) {
-      return file.toPath();
-    }
-  };
-
-  private final ImmutableList.Builder<Path> directories = ImmutableList.builder();
-  private final ImmutableList.Builder<Path> protos = ImmutableList.builder();
+  final List<Path> directories = new ArrayList<>();
+  final List<Path> protos = new ArrayList<>();
 
   /** Open ZIP files that need to be released. Only non-empty while loading. */
-  private final Map<Path, ZipFile> zipFiles = new LinkedHashMap<>();
+  final Map<Path, ZipFile> zipFiles = new LinkedHashMap<>();
 
   /** Add one or more directories from which proto files will be loaded. */
   public SchemaLoader addDirectory(File file) {
     return addDirectory(file.toPath());
   }
 
-  /** Add directories from which proto files will be loaded. */
-  public SchemaLoader addDirectories(Collection<? extends File> files) {
-    return addDirectories(Iterables.transform(files, FILE_TO_PATH));
-  }
-
   /** Add one or more directories from which proto files will be loaded. */
   public SchemaLoader addDirectory(Path path) {
     directories.add(path);
     return this;
   }
 
-  /** Add directories from which proto files will be loaded. */
-  public SchemaLoader addDirectories(Iterable<? extends Path> paths) {
-    directories.addAll(paths);
-    return this;
+  /** Returns a mutable list of the directories that this loader will load from. */
+  public List<Path> directories() {
```

Comment:
This is along the same pattern as OkHttp's interceptors, where a mutable class returns a mutable list. I was playing with the current API and programatically this is more powerful, especially for checks like "do we have _any_ directories?"